lib/tokenize.cpp

```diff
@@ -1019,20 +1019,16 @@ namespace {
     };
 }
 
-bool Tokenizer::isFunctionPointer(const Token* tok) {
-    return Token::Match(tok, "%name% ) (");
-}
-
 void Tokenizer::simplifyTypedef()
 {
     // Simplify global typedefs that are not redefined with the fast 1-pass simplification.
     // Then use the slower old typedef simplification.
-    std::map<std::string, int> numberOfTypedefs;
+    std::map<std::string, std::set<std::string>> numberOfTypedefs;
     for (Token* tok = list.front(); tok; tok = tok->next()) {
         if (tok->str() == "typedef") {
             TypedefSimplifier ts(tok);
             if (!ts.fail())
-                numberOfTypedefs[ts.name()]++;
+                numberOfTypedefs[ts.name()].insert(ts.getTypedefToken()->strAt(1));
```

Comment:
I don't remember the details of this code. But it looks shaky to just pick the next token. can't it be some qualifier or storage specifier like "const" "static" etc. Or "struct", "union", "enum",..